controller/proxycontroller.php

```diff
@@ -65,9 +86,27 @@ public function proxy($url) {
 			]);
 		} catch (ClientException $e) {
 			$error_code = $e->getResponse()->getStatusCode();
-			$response = new JSONResponse();
-			$response->setStatus($error_code);
+			$message = $e->getMessage();
+
+			$this->logger->debug($message);
+			$response = new JSONResponse([
+				'message' => $message,
+				'proxy_code' => $error_code
+			], Http::STATUS_IM_A_TEAPOT);
```

Comment:
Not saying I didn't smile, but maybe there could be a more helping http code here. 🙂